modules/images/webp-uploads/image-edit.php

```diff
@@ -91,15 +113,20 @@ function ( $metadata, $post_meta_id ) use ( $post_id, $file_path, $mime_type, $e
 				return $metadata;
 			}
 			$callback_executed = true;
-
 			// No sizes to be created.
 			if ( empty( $metadata['sizes'] ) ) {
 				return $metadata;
 			}
 
 			$old_metadata = wp_get_attachment_metadata( $post_id );
 			$resize_sizes = array();
+			$target       = isset( $_REQUEST['target'] ) ? $_REQUEST['target'] : 'all';
+
 			foreach ( $old_metadata['sizes'] as $size_name => $size_details ) {
+				if ( 'nothumb' === $target && 'thumbnail' === $size_name ) {
```

Comment:
Let's add a comment to explain:
```suggestion
				// If the target is 'nothumb', skip generating the 'thumbnail' size.
				if ( 'nothumb' === $target && 'thumbnail' === $size_name ) {
```